File: java/v1/src/main/java/com/api/v1/dtos/payment/PaymentResponseDto.java
package com.api.v1.dtos.payment;

import com.api.v1.dtos.creditcard.CreditCardResponseDto;

import java.math.BigInteger;

public record PaymentResponseDto(
        BigInteger orderNumber,
        CreditCardResponseDto creditCard,
        String paidAt
) {
}

File: kotlin/v2/src/main/kotlin/com/api/v2/payment/dtos/PaymentResponseDto.kt
package com.api.v2.payment.dtos

import com.api.v2.creditcard.dtos.CreditCardResponseDto
import java.math.BigInteger

data class PaymentResponseDto(
    val orderNumber: BigInteger,
    val creditCard: CreditCardResponseDto,
    val paidAt: String
)